jetty-client/src/test/java/org/eclipse/jetty/client/ClientConnectionCloseTest.java

```diff
@@ -249,8 +249,7 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
                 DuplexConnectionPool connectionPool = (DuplexConnectionPool)destination.getConnectionPool();
                 HttpConnectionOverHTTP connection = (HttpConnectionOverHTTP)connectionPool.getActiveConnections().iterator().next();
                 assertFalse(connection.getEndPoint().isOutputShutdown());
-            })
-            .onResponseHeaders(r -> r.getHeaders().remove(HttpHeader.CONNECTION));
```

Comment:
So do you want to the signature of `Response.HeaderListener` to be:
```java
boolean onHeader(Response response, HttpField.Mutable field);
```
Currently I have not exposed the mutable for the response headers.

Is it expected that listeners in the client mutate the headers received or is this just for this test?

Reply:
Ah good point. It is just a trick for the test. `HttpClient` detects when there is a `Connection: close` and initiates the close locally. I want to remove the `Connection: close` so that I want to verify that the TCP close by the server triggers the client to TCP close too. However, I can see applications wanting to filter out certain headers (e.g. cookies) so perhaps a `Mutable` is not that bad in this listener.

Reply:
For now I have just down cast in the unit test to access the mutable headers, so the test works.  I'll defer to #4808 for any other changes to the response listener APIs